App.tsx

```diff
@@ -12,8 +13,13 @@ export default function App() {
   <SafeAreaProvider>
       <NavigationContainer>
         <Stack.Navigator screenOptions={{ headerShown: false }}>
-          <Stack.Screen name="Main" component={MainTabs} />
-          <Stack.Screen name="SchedulePage" component={SchedulePage} />
+          <Stack.Group>
+            <Stack.Screen name="Main" component={MainTabs} />
+            <Stack.Screen name="SchedulePage" component={SchedulePage} />
+          </Stack.Group>
+          <Stack.Group screenOptions={{ presentation: 'modal' }}>
```

Comment:
(will stop calling out individual but just mentioning to keep in mind) - this pattern was here but we'll want to extract this to a `const` outside the component too (to avoid re-rendering the screen under)